src/centralserver/registration-service/src/main/java/org/niis/xroad/centralserver/registrationservice/service/AdminApiServiceImpl.java

```diff
@@ -0,0 +1,149 @@
+import org.niis.xroad.centralserver.registrationservice.openapi.model.ManagementRequestType;
+import org.niis.xroad.centralserver.registrationservice.openapi.model.XRoadId;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestClientResponseException;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+
+@Service
+@Slf4j
+class AdminApiServiceImpl implements AdminApiService {
+
+    public static final String REQUEST_FAILED = "Registration request failed";
+    private final RestTemplate restTemplate;
+    private final ObjectMapper mapper;
+
+    @SuppressWarnings("checkstyle:MagicNumber")
+    AdminApiServiceImpl(RegistrationServiceProperties properties, RestTemplateBuilder builder, ObjectMapper mapper) {
+
+        CloseableHttpClient client;
+        try {
+            client = HttpClients.custom()
+                    .setSSLHostnameVerifier(new NoopHostnameVerifier())
+                    .setSSLContext(SSLContexts.custom()
+                            .setProtocol("TLSv1.3")
+                            .loadTrustMaterial(
+                                    properties.getApiTrustStore().toFile(),
+                                    properties.getApiTrustStorePassword().toCharArray())
+                            .build())
+                    .setDefaultRequestConfig(RequestConfig.custom()
+                            .setConnectTimeout(1000)
+                            .setSocketTimeout(5000)
+                            .setConnectionRequestTimeout(10000)
+                            .build())
+                    .disableAutomaticRetries()
+                    .disableCookieManagement()
+                    .disableRedirectHandling()
+                    .setUserAgent("X-Road Registration Service/7")
+                    .build();
+        } catch (NoSuchAlgorithmException | KeyStoreException | KeyManagementException | CertificateException e) {
+            throw new IllegalStateException("Unable to create HTTP clients", e);
+        } catch (IOException e) {
+            throw new UncheckedIOException("Unable to load trust material", e);
+        }
+
+        this.mapper = mapper;
+        this.restTemplate = builder
+                .requestFactory(() -> new HttpComponentsClientHttpRequestFactory(client))
+                .rootUri(properties.getApiBaseUrl().toString())
+                .defaultHeader("Authorization", "X-ROAD-APIKEY TOKEN=" + properties.getApiToken())
+                .build();
```

Comment:
API tokens: token is not currently generated automatically in installation process (and perhaps it cannot be?).

Should we have a backlog item about this? (maybe ping @raits ?)